gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/MongoUserRepository.java

```diff
@@ -89,25 +94,26 @@ public Flowable<User> findByDomainAndEmail(String domain, String email, boolean
         return Flowable.fromPublisher(usersCollection.find(mongoQuery)).map(this::convert);
     }
 
+    @Override
+    public Maybe<User> findById(UserId id) {
+        return findOne(usersCollection, userIdMatches(id), this::convert);
```

Comment:
**question**: I wondering if it is safe to use userIdMatch in the UserRepositories :thinking: I would prefer to stay with a search by id to avoid side effect...

Reply:
i've left calls to userIdMatches, but changed the implementation for user repositories to only compare the internal id